pages/tips/[id].js

```diff
@@ -0,0 +1,21 @@
+import { useRouter } from "next/router";
+import TipDetails from "/components/TipDetails/";
+
+export default function TipDetailsPage({ tips }) {
+  const router = useRouter();
+
+  const { id } = router.query;
+
+  const tip = tips.find((tip) => tip.id === id);
+
+  if (!tip) {
```

Comment:
when you directly visit this page, the router will not be ready, it won't have the id with the first render. Therefore you will get a flash of your error message. You can prevent this by checking if the router is ready, and if not show a blank page:
```jsx
if(!router.isReady) {
  return null;
}
```

Reply:
Thanks for the hint! We will implement that